enumerables.rb

```diff
@@ -0,0 +1,155 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  # Task-1 my_each exercise
+  def my_each
+    if block_given?
+      length.times { |e| yield(self[e]) }
+    else
+      to_enumerator(:my_each)
+    end
+  end
```

Comment:
#my_each returns an Enumerator if no block is given
#my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
#my_each when a block is given when self is a range returns the range itself after calling the 
given block once for each element in self
#my_each when a block is given when self is a range behaves like #each calls the given block once for each element in self
#my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self